### PR TITLE
Add SF100Linux Formula

### PR DESCRIPTION
**Pull request by superjeng1**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/s/sf100linux.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  homepage "https://github.com/DediProgSW/SF100Linux"
+  url "https://github.com/DediProgSW/SF100Linux/archive/refs/tags/V1.14.20.x.tar.gz"
+  sha256 "d3e710c2a4361b7a82e1fee6189e88a6a6ea149738c9cb95409f0a683e90366e"
+  license "GPL-2.0-or-later"
+
+  depends_on "libusb"
+  depends_on :macos
+  depends_on "pkg-config"
+
+  patch :DATA
```
Comment (SMillerDev) [inline Formula/s/sf100linux.rb:12]: This should be submitted upstream

File: Formula/s/sf100linux.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
+class Sf100linux < Formula
+  desc "Linux software for Dediprog SF100 and SF600 SPI flash programmers"
+  homepage "https://github.com/DediProgSW/SF100Linux"
+  url "https://github.com/DediProgSW/SF100Linux/archive/refs/tags/V1.14.20.x.tar.gz"
+  sha256 "d3e710c2a4361b7a82e1fee6189e88a6a6ea149738c9cb95409f0a683e90366e"
+  license "GPL-2.0-or-later"
+
+  depends_on "libusb"
+  depends_on :macos
```
Comment (SMillerDev) [inline Formula/s/sf100linux.rb:9]: Why?

File: Formula/s/sf100linux.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  depends_on "pkg-config"
+
+  patch :DATA
+
+  def install
+    system "make"
+    bin.install "dpcmd"
+    include.install "ChipInfoDb.dedicfg"
```
Comment (SMillerDev) [inline Formula/s/sf100linux.rb:17]: Is there no `make install`?

**Comment by superjeng1** [comment]
#169525 Is the same thing but for Linux. This patch allows the program to be run on macOS. What should I name this formula?

**Comment by SMillerDev** [comment]
> https://github.com/Homebrew/homebrew-core/pull/169525 Is the same thing but for Linux. This patch allows the program to be run on macOS. What should I name this formula?

The patch should be submitted to the original project and then we can just make the other PR work on macOS

**Pull request closed without merging**